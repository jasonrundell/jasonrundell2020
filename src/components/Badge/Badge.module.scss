.root {
  display: inline-flex;
  border-radius: $sizeSmallest;
  background-color: $backgroundColorPrime;
  color: $accentColorPrime;
  width: $sizeLarge;
  height: $sizeLarge;
  line-height: $sizeNormal;
  font-size: $sizeSmall;
  font-weight: 700;
  justify-content: center;
  align-items: center;
}

.inverse {
  background-color: $accentColorPrime;
  color: $backgroundColorPrime;
  border: 1px solid $accentColorPrime;
}

.tight {
  padding: 0.75rem;
}

.roomy {
  padding: 2rem;
}
