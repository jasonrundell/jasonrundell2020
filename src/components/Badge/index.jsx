import React from 'react'
import PropTypes from 'prop-types'

import Layout from '../Layout'

import { classNames } from '../../utils/css'

// CSS
import { root, inverse as inverseStyle } from './Badge.module.scss'

const { Box } = Layout

const Badge = ({ icon, isInverse }) => {
  const classes = classNames(root, isInverse && inverseStyle)
  return (
    <div className={classes}>
      <Box isTight>{icon}</Box>
    </div>
  )
}

Badge.defaultProps = {
  isInverse: false,
}

Badge.propTypes = {
  icon: PropTypes.any.isRequired,
  isInverse: PropTypes.bool,
}

export default Badge
