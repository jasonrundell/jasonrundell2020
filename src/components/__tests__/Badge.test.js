import React from 'react'
import { create } from 'react-test-renderer'

import Badge from '../Badge'

describe('Badge component', () => {
  test('Matches the snapshot', () => {
    const component = create(<Badge icon="Icon" />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
