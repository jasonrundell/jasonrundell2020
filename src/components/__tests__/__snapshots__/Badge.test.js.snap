// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component Badge renders correctly 1`] = `
<div
  className="root"
>
  <div
    className="root tight"
  >
    icon
  </div>
</div>
`;
